pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -1236,15 +1460,132 @@ func (a *ApplicationRestoreController) restoreResources(
+					APIVersion: restore.APIVersion,
+					Kind:       restore.Kind,
+					Namespace:  restore.Namespace,
+					Name:       restore.Name,
+				}
+				backupLocation, err := storkops.Instance().GetBackupLocation(backup.Spec.BackupLocation, backup.Namespace)
+				if err != nil {
+					return fmt.Errorf("error getting backup location path: %v", err)
+				}
+				destination := &kdmpapi.ResourceExportObjectReference{
+					// TODO: .GetBackupLocation is not returning APIVersion and kind.
+					// Hardcoding for now.
+					// APIVersion: backupLocation.APIVersion,
+					// Kind:       backupLocation.Kind,
+					APIVersion: "stork.libopenstorage.org/v1alpha1",
+					Kind:       "BackupLocation",
+					Namespace:  backupLocation.Namespace,
+					Name:       backupLocation.Name,
+				}
+				resourceExport.Spec.Source = *source
+				resourceExport.Spec.Destination = *destination
+				_, err = kdmpShedOps.Instance().CreateResourceExport(resourceExport)
+				if err != nil {
+					logrus.Errorf("failed to create DataExport CR: %v", err)
+					return err
+				}
+				return nil
+			}
+			logrus.Errorf("failed to get restore resourceExport CR: %v", err)
+			// Will retry in the next cycle of reconciler.
+			return nil
+		} else {
+			var message string
+			// Check the status of the resourceExport CR and update it to the applicationBackup CR
+			logrus.Debugf("resource export: %s, status: %s", resourceExport.Name, resourceExport.Status.Status)
+			switch resourceExport.Status.Status {
+			case kdmpapi.ResourceExportStatusFailed:
+				message = fmt.Sprintf("Error applying resources: %v", err)
+				restore.Status.Status = storkapi.ApplicationRestoreStatusFailed
+				restore.Status.Stage = storkapi.ApplicationRestoreStageFinal
+				restore.Status.Reason = message
+				restore.Status.LastUpdateTimestamp = metav1.Now()
+				err = a.client.Update(context.TODO(), restore)
+				if err != nil {
+					return err
+				}
+				a.recorder.Event(restore,
+					v1.EventTypeWarning,
+					string(storkapi.ApplicationRestoreStatusFailed),
+					message)
+				log.ApplicationRestoreLog(restore).Errorf(message)
+				return err
+			case kdmpapi.ResourceExportStatusSuccessful:
+				// Modify to have subresource level updating
+				for _, resource := range resourceExport.Status.Resources {
+					a.updateResourceStatusFromRestoreCR(
+						restore,
+						resource,
+						resource.Status,
+						resource.Reason)
+				}
+				restore.Status.Stage = storkapi.ApplicationRestoreStageFinal
+				restore.Status.FinishTimestamp = metav1.Now()
+				restore.Status.Status = storkapi.ApplicationRestoreStatusSuccessful
+				restore.Status.Reason = "Volumes and resources were restored up successfully"
+
+			case kdmpapi.ResourceExportStatusInitial:
+				doCleanup = false
+			case kdmpapi.ResourceExportStatusPending:
+				doCleanup = false
+			case kdmpapi.ResourceExportStatusInProgress:
+				restore.Status.LastUpdateTimestamp = metav1.Now()
+				doCleanup = false
```

Comment:
We need to update last updatetimestamp in initial, pending, in-progress status to avoid timeout.